src/cmdline/mod.rs

```diff
@@ -219,49 +159,20 @@ impl Cmdline {
         self.insert_str(kv_str)
     }
 
-    /// Validates and inserts a string to the end of the current command line.
+    /// Validates and inserts a string to the end of the current args string.
     ///
     /// # Arguments
     ///
-    /// * `slug` - String to be appended to the command line.
-    ///
-    /// # Examples
-    ///
-    /// ```rust
-    /// # use linux_loader::cmdline::*;
-    /// # use std::ffi::CString;
-    /// let mut cl = Cmdline::new(100);
-    /// cl.insert_str("foobar");
-    /// let cl_cstring = CString::new(cl).unwrap();
-    /// assert_eq!(cl_cstring.to_str().unwrap(), "foobar");
-    /// ```
-    pub fn insert_str<T: AsRef<str>>(&mut self, slug: T) -> Result<()> {
-        let s = slug.as_ref();
-        valid_str(s)?;
-
-        self.has_capacity(s.len())?;
-
-        self.start_push();
-        self.line.push_str(s);
-        self.end_push();
-
-        Ok(())
-    }
+    /// * `slug` - String to be appended to the args string.
+    fn insert_str<T: AsRef<str>>(&mut self, slug: T) -> Result<()>;
+}
 
-    /// Returns the string representation of the command line without the nul terminator.
-    ///
-    /// # Examples
-    ///
-    /// ```rust
-    /// # use linux_loader::cmdline::*;
-    /// let mut cl = Cmdline::new(10);
-    /// cl.insert_str("foobar");
-    /// assert_eq!(cl.as_str(), "foobar");
-    /// ```
-    pub fn as_str(&self) -> &str {
-        self.line.as_str()
-    }
+/// An abstraction for the boot arguments of a kernel command line.
+#[derive(Clone, Debug)]
```

Comment:
We should follow the API guidelines and derive the other useful traits as well. Here are the Rust guidelines: https://rust-lang.github.io/api-guidelines/interoperability.html

Reply:
Adding implementation for Eq, PartialEq, Ord, PartialOrd, Hash, Default and Dsiplay traits as well for both BootArgs and InitArgs. Same traits where implemented for Cmdline, most of them using #derive, and a few of them by providing actual implementation.